Modules/posixmodule.c

```diff
@@ -12207,11 +12168,20 @@ DirEntry_from_find_data(path_t *path, WIN32_FIND_DATAW *dataW)
     }
 
     find_data_to_file_info(dataW, &file_info, &reparse_tag);
-    _Py_attribute_data_to_stat(&file_info, reparse_tag, &entry->win32_lstat);
+    Py_BEGIN_ALLOW_THREADS
```

Comment:
IMO, a `gil_held` parameter would be better. We only need to release the GIL if it's held around the `GetVolumePathName` call. We don't want to do it for every file. We don't even need to make this call if the reparse tag is zero. Even if it's non-zero, using junctions as links is uncommon, but we could be working with thousands of symlinks or non-link reparse points.